std/browser/_ops_shim.ts

```diff
@@ -0,0 +1,289 @@
+// Copyright 2018-2020 the Deno authors. All rights reserved. MIT license.
+
+import { errors } from "https://raw.githubusercontent.com/denoland/deno/v1.1.0/cli/js/errors.ts";
+export { errors } from "https://raw.githubusercontent.com/denoland/deno/v1.1.0/cli/js/errors.ts";
+
+export enum SeekMode {
+  Start = 0,
+  Current = 1,
+  End = 2,
+}
+
+class Resource {
+  get buf(): ArrayBuffer {
+    if (files.has(this.name)) {
+      return files.get(this.name)!.buf;
+    }
+    const buf = new ArrayBuffer(0);
+    files.set(this.name, { buf });
+    return buf;
+  }
+
+  set buf(buf: ArrayBuffer) {
+    if (files.has(this.name)) {
+      files.get(this.name)!.buf = buf;
+    } else {
+      files.set(this.name, { buf });
+    }
+  }
+
+  pos = 0;
+  name: string;
+  options?: Deno.OpenOptions;
+  closed = false;
+
+  constructor(name: string) {
+    this.name = name;
+  }
+}
+
+interface VirtualFile {
+  buf: ArrayBuffer;
+}
+
+const files = new Map<string, VirtualFile>();
+
+const resources = new Map<number, Resource>();
+let resourceId = 0;
+
+function closeResource(rid: number): void {
+  if (resources.has(rid)) {
+    resources.get(rid)!.closed = true;
```

Comment:
Could a resource not be removed from the resource table after it has been closed? If I understand correctly the contents would still persist in the `files` table. That way you don't need the `closed` attribute on `Resource`, and don't need to check for its truthyness all over the place.